### PR TITLE
setUserPreferences

### PR DESCRIPTION
**Pull request by mbasadi**
setUserPreferences

### CODE REVIEW

**Comment by mbasadi** [comment]
[Update Back-end SDK](https://trello.com/c/6NrSqBQK/568-update-back-end-sdk)

File: src/__tests__/index.test.ts [hunk 1]
```diff
@@ -400,3 +401,53 @@ describe('deleteSubNotification by subNotificationId', () => {
...
+        state: true,
+        subNotificationId: 'subNotificationId'
+      }
+    ];
+    test('makes API calls to the correct end-point', async () => {
+      axiosMock.onPost(retractEndPointRegex).reply(200);
+      await notificationapi.init(clientId, clientSecret);
+      await notificationapi.setUserPreferences(userId, userPreferences);
+      expect(axiosMock.history.post).toHaveLength(1);
+      expect(axiosMock.history.post[0].url).toEqual(
```
Comment (sahandseifi) [inline src/__tests__/index.test.ts:445]: With subNotificationId, you are probably not too concerned with the URL. You probably just want to check that the parameter is sent correctly. So feel free to simplify the test to only check what you intend to check.

**Review by sahandseifi — CHANGES_REQUESTED** [review]

**Comment by mbasadi** [comment]
@sahandseifi Thanks for your comments. Please check again.

File: src/__tests__/index.test.ts [hunk 2]
```diff
@@ -400,3 +401,54 @@ describe('deleteSubNotification by subNotificationId', () => {
...
+  ];
+  test('makes API calls to the correct end-point', async () => {
+    axiosMock.onPost(retractEndPointRegex).reply(200);
+    await notificationapi.init(clientId, clientSecret);
+    await notificationapi.setUserPreferences(userId, userPreferences);
+    expect(axiosMock.history.post).toHaveLength(1);
+    expect(axiosMock.history.post[0].url).toEqual(
+      `https://api.notificationapi.com/${clientId}/user_preferences/${userId}`
+    );
+    expect(axiosMock.history.post[0].data).toEqual(
```
Comment (sahandseifi) [inline src/__tests__/index.test.ts:425]: When asserting completely different things, separate test cases.

Check the retract tests

File: src/interfaces.ts [hunk 3]
```diff
@@ -36,6 +36,12 @@ export interface DeleteSubNotificationRequest {
   notificationId: string;
   subNotificationId: string;
 }
+export interface SetUserPreferences {
```
Comment (sahandseifi) [inline src/interfaces.ts:39]: SetUserPreferences sounds like the name of a function. Please rename.

**Review by sahandseifi — CHANGES_REQUESTED** [review]

**Pull request merged**